### PR TITLE
fixes #5506 - added scoped search to SmartProxy for UI and API v2

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/controllers/smart_proxies_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class SmartProxiesController < ApplicationController
   before_filter :find_by_name, :only => [:edit, :update, :refresh, :ping, :destroy]
 
   def index
-    @smart_proxies = resource_base.includes(:features).paginate :page => params[:page]
+    @smart_proxies = resource_base.includes(:features).search_for(params[:search], :order => params[:order]).paginate(:page => params[:page])
```
Comment (isratrade) [inline app/controllers/smart_proxies_controller.rb:7]: Surprisingly there was previously no scoped search in the UI

**Comment by isratrade** [comment]
@witlessbird, @abenari  I would like to make search by feature case-insenstive,so `feature = dns` works. Currently, I need to do `feature = DNS` or `feature ~ dns`

**Comment by isratrade** [comment]
I will add some functional tests soon.

**Comment by isratrade** [comment]
added functional tests

**Comment by abenari** [comment]
@isratrade in did scoped-search is case sensitive for the equal (=) operator and case insensitive for the like (~) operator. This is true across all databases and can't be modified via scoped search options.
I am sure the scoped search community will welcome a patch that allows modifying the case behaviour :)

**Comment by dmitri-d** [comment]
It's probably worth it to use all lower-case for SmartProxy feature names...

**Comment by dmitri-d** [comment]
lgtm, although I didn't test the PR...

**Comment by isratrade** [comment]
all tests pass. Only failure is known  random 'rundeck' failure

**Comment by domcleal** [comment]
@isratrade when you say you'd like to make this case insensitive, did you mean now (before merge) or one day?

**Comment by isratrade** [comment]
@domcleal, I thought if scoped search can pass a variable to make the search case insensitve, but `~` instead of `=` is the answer.  There is no need to change anything (from my side) to merge it.

**Comment by domcleal** [comment]
@isratrade cool, fine by me.

One last thing I noticed, could you add a scoped search definition on URL please?

**Comment by isratrade** [comment]
updated. `url` added to scope search

**Comment by domcleal** [comment]
Merged as 428b1cf for Foreman 1.5.0, thanks @isratrade!

**Pull request closed without merging**